jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -41,6 +43,8 @@
   /** Helper for loading a .class file. */
   private static class ClassFileLoader extends ClassLoader {
 
+    private static final Map<Path, Class> cache = new HashMap<>();
```

Comment:
We should be careful with global states like this since having the cache be a static variable means that the `ClassFileLoader` is now not thread-safe. This would mean that we would have to make sure we don't use this `MainClassFinder` in multiple threads concurrently, or we change this cache to be an instance field, or use a thread-safe map.

Reply:
Should I use a `ConcurrentHashMap`?